### ISSUE TITLE
Reproducing the result on ImageNet

### ISSUE THREAD

**Issue by FriedRonaldo** [snu-mllab/PuzzleMix#6]
Hi, thanks for the interesting work and its codes.

I have a question about the reproducibility of the work. Recently, I ran the provided code to reproduce the accuracy on the ImageNet. (run_fast.sh)
However, the code with Apex outputs a loss of NaN. After removing Apex, the loss of NaN does not appear. 
So I ran the code w/o Apex, however, after the training, the best accuracy of ResNet-50 with PuzzleMix is 75.63 that is significantly lower than 77.51 shown in the paper. 

Without Apex, I did not change anything. 
* I also created the resized dataset (160, 352). 
* I used 4 * V100 GPUs.
* pytorch version is 1.2.0

* In this case, how can I reproduce the result of the paper? 
* Is it significant to use the mixed-precision strategy? (As I searched, the improvement by the mixed-precision is lower than 1%p.)

Thanks again.

Best,
Kyungjune, Baek

**Comment by Janghyun1230** [comment]
Hello! I feel sorry that I haven't tested the effect of Apex. However, I've trained the model without Apex for 300 epochs following CutMix setting (see https://github.com/snu-mllab/PuzzleMix/tree/master/cutmix).

I just ran the code after your issue, but however in my setting, NaN does not occur until 10 epochs where the learning rate is the highest. (test setting: Pytorch 1.7.0, CUDA 11.1, RTX-2080 Ti). Does your NaN occur after the initial 10 epochs of the training?

It seems that there is an issue with Apex in your setting and device. Could you reproduce Vanilla model with apex? Or you can try to run codes with the more recent version of Pytorch (e.g., 1.7) and the corresponding CUDA as Apex recommends the latest stable release, obtainable from https://pytorch.org/ (see https://github.com/NVIDIA/apex).

However, if you need the trained models in our ImageNet experiments I can provide you. 

Best, 
Jang-Hyun

**Comment by FriedRonaldo** [comment]
Thanks for the reply!

* Does your NaN occur after the initial 10 epochs of the training?
    * NaN occurs at the first epoch.
* Could you reproduce Vanilla model with apex?
    * I am sorry that I cannot understand the meaning exactly. I ran your code w/o PuzzleMix but it also raises NaN.

As your recommendation, I will try the code with the latest version of Pytorch.

Actually, we need to train a model from scratch because we want to improve the overall performance of PuzzleMix, however, if you can provide the pretrained model trained under the setting of Table 5. (100epochs setting)

Thanks!

**Comment by wjun0830** [comment]
May I ask for training time for Imagenet on ResNet50 without apex? I mean the one that follows the training protocol from CutMix.

**Comment by FriedRonaldo** [comment]
@wjun0830 
For CutMix protocol, apex is not used by default and I do not know the exact training time of PuzzleMix. Because I added my module to train a model. 
However, roughly speaking, PuzzleMix might take about two weeks with 4xV100 GPU (4*32GB VRAM).
(I hope that the author provides the exact one.)

**Comment by wjun0830** [comment]
> @wjun0830
> For CutMix protocol, apex is not used by default and I do not know the exact training time of PuzzleMix. Because I added my module to train a model.
> However, roughly speaking, PuzzleMix might take about two weeks with 4xV100 GPU (4*32GB VRAM).
> (I hope that the author provides the exact one.)

Thanks for fast reply!
